File: PDM/PDMplugin/src/main/java/com/prolambda/pdm/PDMplugin/SourceCode.java
package com.prolambda.pdm.PDMplugin;

import com.thoughtworks.xstream.annotations.XStreamAlias;

@XStreamAlias("dependence")
public class SourceCode {
	
	@XStreamAlias("url")
	private String url;
	
	@XStreamAlias("version")
	private String version;
	public String getUrl() {
		return url;
	}
	public void setUrl(String url) {
		this.url = url;
	}
	public String getVersion() {
		return version;
	}
	public void setVersion(String version) {
		this.version = version;
	}
}

File: PDM/PDMBuilder/PDMBuilder/SourceCode.cpp
#include "StdAfx.h"
#include "SourceCode.h"


SourceCode::SourceCode()
{
}


SourceCode::~SourceCode()
{
}

SourceCode::SourceCode(SourceCode &sourceCode){
	this->url = sourceCode.url;
	this->version = sourceCode.version;
	this->localPath = sourceCode.localPath;
}

string SourceCode::GetUrl()
{
	return url;
}

void SourceCode::SetUrl(string url){
	this->url = url;
}

string SourceCode::GetVersion()
{
	return version;
}

void SourceCode::SetVersion(string version)
{
	this->version = version;
}

string SourceCode::GetLocalPath()
{
	return localPath;
}

void SourceCode::SetLocalPath(string localPath)
{
	this->localPath = localPath;
}